src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6952,17 +7049,18 @@ private ImmutableArray<VisitResult> VisitArguments(
                 }
 
                 // Re-infer method type parameters
-                if (method?.IsGenericMethod == true)
+                if (method?.GetMemberArityIncludingExtension() > 0)
                 {
                     if (HasImplicitTypeArguments(node))
                     {
                         method = InferMethodTypeArguments(method, GetArgumentsForMethodTypeInference(results, argumentsNoConversions), refKindsOpt, argsToParamsOpt, expanded);
-                        parametersOpt = method.Parameters;
+                        parametersOpt = method.GetParametersIncludingExtensionParameter();
```

Comment:
>parametersOpt = method.GetParametersIncludingExtensionParameter(); [](http://example.com/codeflow?start=24&length=66)

Are we testing this code path for static extensions?

Reply:
Yes, `Nullability_Invocation_09`, the `object.M2()` invocation